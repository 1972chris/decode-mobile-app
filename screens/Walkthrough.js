import React from 'react';
import { Image } from 'react-native';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Onboarding from 'react-native-onboarding-swiper';
import { goToPinSetup } from '../application/redux/actions/navigation';
import TitleElement from '../application/components/TitleElement/TitleElement';
import DoneButton from '../application/components/DoneButton/DoneButton';

const firstPageImage = require('../assets/images/onboarding01.png');
const secondPageImage = require('../assets/images/onboarding02.png');
const thirdPageImage = require('../assets/images/onboarding03.png');
const fourthPageImage = require('../assets/images/onboarding04.png');


const subtitleStyle = {
  flex: 4,
  fontSize: 32,
  textAlign: 'center',
  color: 'white',
  paddingLeft: 35,
  paddingRight: 35,
};

const titleStyle = {
  flex: 1,
  fontSize: 40,
  color: 'white',
};

const Walkthrough = props => (
  <Onboarding
    onSkip={() => props.goToPinSetup()}
    DoneButtonComponent={() => (
      <DoneButton onPress={() => props.goToPinSetup()} />
    )}
    bottomBarHighlight={false}
    skipLabel="Skip"
    nextLabel="Next"
    pages={[
      {
        imageContainerStyles: { flex: 7 },
        backgroundColor: '#0575e5',
        image: <Image style={{ width: '95%', height: 400 }} resizeMode="contain" source={firstPageImage} />,
        title: <TitleElement title={'Decode'.toUpperCase()} style={titleStyle} />,
        subtitle: <TitleElement title="Gives you ownership of your personal data" style={subtitleStyle} />,
      },
      {
        imageContainerStyles: { flex: 7 },
        backgroundColor: '#fd56ac',
        image: <Image style={{ width: '95%', height: 400 }} resizeMode="contain" source={secondPageImage} />,
        title: '',
        subtitle: <TitleElement title="Big companies store and sell your data" style={subtitleStyle} />,
      },
      {
        imageContainerStyles: { flex: 7 },
        backgroundColor: '#37bdc2',
        image: <Image style={{ width: '95%', height: 400 }} resizeMode="contain" source={thirdPageImage} />,
        title: <TitleElement title={'Decode'.toUpperCase()} style={titleStyle} />,
        subtitle: <TitleElement title="Gives you back control of your data. It is kept in your wallet" style={subtitleStyle} />,
      },
      {
        imageContainerStyles: { flex: 7 },
        backgroundColor: '#0475e5',
        image: <Image style={{ width: '95%', height: 400 }} resizeMode="contain" source={fourthPageImage} />,
        title: '',
        subtitle: <TitleElement title="Your data is protected" style={subtitleStyle} />,
      },
    ]}
  />
);

Walkthrough.propTypes = {
  goToPinSetup: PropTypes.func.isRequired,
};

const mapStateToProps = () => ({});

const mapDispatchToProps = dispatch => ({
  goToPinSetup: () => dispatch(goToPinSetup()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Walkthrough);
