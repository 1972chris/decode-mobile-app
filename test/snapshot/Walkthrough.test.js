import React from 'react';
import configureStore from 'redux-mock-store';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import thunk from 'redux-thunk';
import Walkthrough from '../../screens/Walkthrough';

jest.mock('../../node_modules/ex-react-native-i18n', () => ({
  locales: { get: () => ({}) },
  t: () => 'Skip',
}));

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore([thunk]);

const initialState = {
  navigation: {},
};

it('renders Walkthrough component', () => {
  const wrapper = shallow(
    <Walkthrough />,
    { context: { store: mockStore(initialState) } },
  );
  expect(wrapper.dive()).toMatchSnapshot();
});
