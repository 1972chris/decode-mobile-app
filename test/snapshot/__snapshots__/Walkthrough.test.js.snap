// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Walkthrough component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Walkthrough
    goToPinSetup={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "DoneButtonComponent": [Function],
      "DotComponent": [Function],
      "NextButtonComponent": [Function],
      "SkipButtonComponent": [Function],
      "bottomBarHeight": 60,
      "bottomBarHighlight": true,
      "controlStatusBar": true,
      "nextLabel": "Next",
      "onDone": [Function],
      "onSkip": [Function],
      "pages": Array [
        Object {
          "backgroundColor": "#0575e5",
          "image": <Image
            resizeMode="contain"
            source={1}
            style={
              Object {
                "height": 400,
                "width": "95%",
              }
            }
          />,
          "imageContainerStyles": Object {
            "flex": 7,
          },
          "subtitle": <TitleElement
            style={
              Object {
                "color": "white",
                "flex": 7,
                "fontSize": 36,
                "paddingLeft": 50,
                "paddingRight": 50,
                "textAlign": "center",
              }
            }
            title="Gives you ownership of your personal data"
          />,
          "title": <TitleElement
            style={
              Object {
                "color": "white",
                "flex": 1,
                "fontSize": 48,
              }
            }
            title="DECODE"
          />,
        },
        Object {
          "backgroundColor": "#fe6e58",
          "image": <Image
            resizeMode="contain"
            source={1}
            style={
              Object {
                "height": 400,
                "width": "100%",
              }
            }
          />,
          "subtitle": "This is the subtitle that sumplements the title.",
          "title": "The Title",
        },
        Object {
          "backgroundColor": "#999",
          "image": <Image
            resizeMode="contain"
            source={1}
            style={
              Object {
                "height": 400,
                "width": "100%",
              }
            }
          />,
          "subtitle": "Beautiful, isn't it?",
          "title": "Triangle",
        },
        Object {
          "backgroundColor": "#999",
          "image": <Image
            resizeMode="contain"
            source={1}
            style={
              Object {
                "height": 400,
                "width": "100%",
              }
            }
          />,
          "subtitle": "Beautiful, isn't it?",
          "title": "Triangle",
        },
      ],
      "showDone": true,
      "showNext": true,
      "showSkip": true,
      "skipLabel": "Skip",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "DoneButtonComponent": [Function],
        "DotComponent": [Function],
        "NextButtonComponent": [Function],
        "SkipButtonComponent": [Function],
        "bottomBarHeight": 60,
        "bottomBarHighlight": true,
        "controlStatusBar": true,
        "nextLabel": "Next",
        "onDone": [Function],
        "onSkip": [Function],
        "pages": Array [
          Object {
            "backgroundColor": "#0575e5",
            "image": <Image
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "height": 400,
                  "width": "95%",
                }
              }
            />,
            "imageContainerStyles": Object {
              "flex": 7,
            },
            "subtitle": <TitleElement
              style={
                Object {
                  "color": "white",
                  "flex": 7,
                  "fontSize": 36,
                  "paddingLeft": 50,
                  "paddingRight": 50,
                  "textAlign": "center",
                }
              }
              title="Gives you ownership of your personal data"
            />,
            "title": <TitleElement
              style={
                Object {
                  "color": "white",
                  "flex": 1,
                  "fontSize": 48,
                }
              }
              title="DECODE"
            />,
          },
          Object {
            "backgroundColor": "#fe6e58",
            "image": <Image
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "height": 400,
                  "width": "100%",
                }
              }
            />,
            "subtitle": "This is the subtitle that sumplements the title.",
            "title": "The Title",
          },
          Object {
            "backgroundColor": "#999",
            "image": <Image
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "height": 400,
                  "width": "100%",
                }
              }
            />,
            "subtitle": "Beautiful, isn't it?",
            "title": "Triangle",
          },
          Object {
            "backgroundColor": "#999",
            "image": <Image
              resizeMode="contain"
              source={1}
              style={
                Object {
                  "height": 400,
                  "width": "100%",
                }
              }
            />,
            "subtitle": "Beautiful, isn't it?",
            "title": "Triangle",
          },
        ],
        "showDone": true,
        "showNext": true,
        "showSkip": true,
        "skipLabel": "Skip",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
